nucypher/characters/chaotic.py

```diff
@@ -0,0 +1,263 @@
+from nucypher.cli.types import ChecksumAddress
+from nucypher.crypto.powers import ThresholdRequestDecryptingPower
+from nucypher.network.decryption import ThresholdDecryptionClient
+from nucypher.policy.conditions.types import LingoList
+from nucypher.policy.conditions.utils import validate_condition_lingo
+
+
+class FakeNode:
+    def __init__(self, checksum_address):
+        self.checksum_address = checksum_address
+
+
+class _ParticipantKeyDict(dict):
+    def __init__(self, threshold_request_decrypting_power, *args, **kwargs):
+        self.threshold_request_decrypting_power = threshold_request_decrypting_power
+        super().__init__(*args, **kwargs)
+
+    def __getitem__(self, _item):
+        # Everybody has the same public key at the moment.
+        fifty_fiver = (
+            self.threshold_request_decrypting_power._get_static_secret_from_ritual_id(
+                55
+            )
+        )
+        return fifty_fiver.public_key()
+
+
+class _Blasphemy:
+    def __init__(
+        self, tau, threshold, shares_num, checksum_addresses, session_seed=None
+    ):
+        self.tau = tau
+        self.threshold = threshold
+        self.shares = shares_num
+        self.fake_nodes = [
+            FakeNode(checksum_address) for checksum_address in checksum_addresses
+        ]
+        self.checksum_addresses = checksum_addresses
+        if session_seed == None:
+            session_seed = b"ABytestringOf32BytesIsNeededHere"
+        secret_factory = SessionSecretFactory.from_secure_randomness(session_seed)
+        self.threshold_request_decrypting_power = ThresholdRequestDecryptingPower(
+            session_secret_factory=secret_factory
+        )
+        self.participant_public_keys = _ParticipantKeyDict(
+            self.threshold_request_decrypting_power
+        )
+
+
+class DKGOmniscient:
+    """
+    A mixin to give a Character the ability to see the inner-workings of a DKG.
+    """
+
+    class DKGInsight:
+        # TODO: Make these configurable:
+        tau = 1
+        security_threshold = 3
+        shares_num = 4
+
+        def __init__(self):
+            """ """
+
+            def gen_eth_addr(i: int) -> str:
+                return f"0x{i:040x}"
+
+            checksum_addresses = [gen_eth_addr(i) for i in range(0, self.shares_num)]
+
+            self.fake_ritual = _Blasphemy(
+                tau=self.tau,
+                threshold=self.security_threshold,
+                shares_num=self.shares_num,
+                checksum_addresses=checksum_addresses,
+            )
+
+            validator_keypairs = [
+                ferveo.Keypair.random() for _ in range(0, self.shares_num)
+            ]
+
+            validators = [
+                ferveo.Validator(checksum_addresses[i], keypair.public_key())
+                for i, keypair in enumerate(validator_keypairs)
+            ]
+
+            # Validators must be sorted by their public key
+            validators.sort(key=lambda v: v.address)
```

Comment:
Why use a lambda when there's a perfectly good stdlib (`operator` module) function to do this?
```suggestion
            validators.sort(key=attrgetter("address")
```